php/commands/src/CLI_Alias_Command.php

```diff
@@ -0,0 +1,335 @@
+	 *
+	 * <key>
+	 * : Key for the alias.
+	 *
+	 * <login>
+	 * : SSH Login.
+	 *
+	 * <wp_path>
+	 * : WordPress install path.
+	 *
+	 * [--config=<config>]
+	 * : Config file to be considered for operations.
+	 * ---
+	 * options:
+	 *   - global
+	 *   - project
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Update alias.
+	 *     $ wp cli alias update @prod newlogin@host /updated/wordpress/path/
+	 *     Success: Updated 'prod' alias.
+	 *
+	 *     # Update project alias.
+	 *     $ wp cli alias update @prod newlogin@host /updated/wordpress/path/ --config=project
+	 *     Success: Updated 'prod' alias.
+	 *
+	 * @subcommand update
+	 */
+	public function update( $args, $assoc_args ) {
+
+		list( $alias, $login, $wp_path ) = $args;
+
+		$config = ( ! empty( $assoc_args['config'] ) ? $assoc_args['config'] : '' );
+
+		list( $config_path, $aliases ) = $this->_get_aliases_data( $config, $alias );
+
+		$this->_validate_config_file( $config_path );
+
+		if ( ! empty( $aliases[ $alias ] ) ) {
+			$aliases[ $alias ]['ssh'] = $login . ':' . $wp_path;
+			$yaml_data                = Spyc::YAMLDump( $aliases );
+
+			if ( file_put_contents( $config_path, $yaml_data ) ) {
+				WP_CLI::success( sprintf( 'Updated \'%s\' alias.', $alias ) );
+			}
+		} else {
+			WP_CLI::error( sprintf( 'No alias found with key \'%s\'.', $alias ) );
+		}
+	}
+
+	/**
+	 * Get config path and aliases data based on config type.
+	 *
+	 * @param string $config Type of config to get data from.
+	 * @param string $alias  Alias to be used for Add/Update/Delete.
+	 *
+	 * @return array Config Path and Aliases in it.
+	 */
+	private function _get_aliases_data( $config, $alias ) {
+
+		$global_config_path = WP_CLI::get_runner()->get_global_config_path();
+		$global_aliases     = Spyc::YAMLLoad( $global_config_path );
+
+		$project_config_path = WP_CLI::get_runner()->get_project_config_path();
+		$project_aliases     = Spyc::YAMLLoad( $project_config_path );
+
+		if ( 'global' === $config ) {
+			$config_path = $global_config_path;
+			$aliases     = $global_aliases;
+		} elseif ( 'project' === $config ) {
+			$config_path = $project_config_path;
+			$aliases     = $project_aliases;
+		} else {
+
+			$is_global_alias  = array_key_exists( $alias, $global_aliases );
+			$is_project_alias = array_key_exists( $alias, $project_aliases );
+
+			if ( $is_global_alias && $is_project_alias ) {
+				WP_CLI::error( sprintf( 'Key \'%s\' found in more than one path. Please pass --config param.', $alias ) );
+			} elseif ( $is_global_alias ) {
+				$config_path = $global_config_path;
+				$aliases     = $global_aliases;
+			} else {
+				$config_path = $project_config_path;
+				$aliases     = $project_aliases;
+			}
+		}
+
+		return array( $config_path, $aliases );
+
+	}
+
+	/**
+	 * Check if the config file exists and is writable.
+	 *
+	 * @param string $config_path Path to config file.
+	 *
+	 * @return void
+	 */
+	private function _validate_config_file( $config_path ) {
+		try {
```

Comment:
Why throw and catch exceptions here? You can just immediately use `WP_CLI::error()` instead.